srml/staking/src/lib.rs

```diff
@@ -16,7 +16,157 @@
+//! - Once a new era is triggered, rewards are paid to the validators and the associated nominators. 
+//! - The validator can declare an amount, named `validator_payment`, that does not get shared with the nominators at each reward payout through their `ValidatorPrefs`. This value gets deducted from the total reward that can be paid. The remaining portion is split among the validator and all of the nominators who had a vote for this validator, proportional to their staked value. 
+//! - All entities who receive a reward have the option to choose their reward destination, through the `Payee` storage (see `set_payee()`), to be one of the following: 
+//! - Controller account.
+//! - Stash account, not increasing the staked value.
+//! - Stash account, also increasing the staked value.
+//! 
+//! ### Slashing details 
+//! 
+//! - A validator can be _reported_ to be offline at any point via `on_offline_validator` public function. 
+//! - Each validator declares how many times they can be _reported_ before it actually gets slashed via the `unstake_threshold` in `ValidatorPrefs`. On top of this, the module also introduces a `OfflineSlashGrace`, which applies to all validators and prevents them from getting immediately slashed.
+//! - Similar to the reward value, the slash value is updated at the end of each era by multiplying `slot_stake` and a configuration storage item, `OfflineSlash`.
+//! - Once a validator has been reported a sufficient amount of times, the actual value that gets deducted from that validator, and every single nominator that voted for it calculated by multiplying the result of the above point by `2.pow(unstake_threshold)`.
+//! - If the previous overflow, then `slot_stake` is used.
+//! - If the previous is more than what the validator/nominator has in stake, all of their stake is slashed (`.max(total_stake)` in other words).
+//! 
+//! ### Additional Fund Management Operations
+//! 
+//! Any funds already placed into stash can be the target of the following operations:
+//! 
+//! - The controller account can free an portion (or all) of the funds using the `unbond()` call. Note that the funds are not immediately accessible, instead, a duration denoted by `BondingDuration` number of eras must pass until the funds can be actually removed.
+//! - To actually remove the funds, once the bonding duration is over, the `withdraw_unbonded()` can be used.
+//! - As opposed to the above, additional funds can be added to the stash account via the `bond_extra()` transaction call. 
+//! 
+//! ### Election algorithm details.
+//! 
+//! Current election algorithm is implemented based on Phragmén. The reference implementation can be found [here](https://github.com/w3f/consensus/tree/master/NPoS).
+//! 
+//! ## GenesisConfig
+//! 
+//! See the [`GensisConfig`](https://crates.parity.io/srml_staking/struct.GenesisConfig.html) for a list of attributed that can be provided.
```

Comment:
```suggestion
//! See the [`GensisConfig`] for a list of attributed that can be provided.
```

This should be enough, rustdoc supports intra doc links.